dll/win32/shell32/CShellLink.cpp

```diff
@@ -882,6 +882,19 @@ HRESULT STDMETHODCALLTYPE CShellLink::Save(IStream *stm, BOOL fClearDirty)
     m_Header.dwSize = sizeof(m_Header);
     m_Header.clsid = CLSID_ShellLink;
 
+    /* Store target attributes */
+    WIN32_FIND_DATAW wfd = {};
+    WCHAR FsTarget[MAX_PATH];
+    if (GetPath(FsTarget, _countof(FsTarget), NULL, 0) == S_OK && PathFileExistsW(FsTarget))
+    {
+        HANDLE hFind = FindFirstFileW(FsTarget, &wfd);
+        if (hFind != INVALID_HANDLE_VALUE)
+            FindClose(hFind);
+    }
+    C_ASSERT(FIELD_OFFSET(SHELL_LINK_HEADER, ftCreationTime) - 4 == FIELD_OFFSET(SHELL_LINK_HEADER, dwFileAttributes));
+    CopyMemory(&m_Header.dwFileAttributes, &wfd.dwFileAttributes, sizeof(UINT) + (sizeof(FILETIME) * 3));
```

Comment:
Both these structs are set in stone forever and there is no padding because FILETIME is [not aligned as a 64-bit type](https://web.archive.org/web/20190108060654/https://blogs.msdn.microsoft.com/oldnewthing/20040825-00/?p=38053).

Reply:
I don't mind copying each field, trading speed/size for readability is OK by me in this case. I checked and viewing a shell folder with shortcuts in it will not ask for these attributes so speed is not critical. Any other code path does not matter because it will just be for a single shortcut and not a huge list.

Reply:
Chances are that the compiler will optimize it to a memcpy anyway (but an instruction, not a function).

When writing code, we should not try to 'optimize' things like this, if it needs optimization that should be pointed out by a profiler, in which case there should be a comment indicating that it is a critical path and that extensive profiling pointed out that this was a significant improvement.